### ISSUE TITLE
pyodide_build buildpkg does not install Cython as a build dependency when it is spell with a lower case c

### ISSUE THREAD

**Issue by ocefpaf** [pyodide/pyodide#2503]
## 🐛 Bug

When trying to build [cftime](https://github.com/Unidata/cftime) the isolated env does not install cython.

### To Reproduce

`python -m pyodide_build buildpkg packages/cftime/meta.yaml` on [this meta.yaml](https://gist.github.com/ocefpaf/8b9a90bfa40d7dc27c63e3bf22ef335a)

### Expected behavior

Successful build :smile: 

### Environment

- Pyodide Version:
- Browser version:
- Any other relevant information:

### Additional context

A patch to rename `cython` to `Cython` in the cftime pyproject.toml fixed it but we should not be case sensitive with PyPI names.

xref.: https://github.com/pyodide/pyodide/pull/2504

**Comment by hoodmane** [comment]
Thanks for the report!

I think this line needs to change:
https://github.com/pyodide/pyodide/blob/d74bc74c9d76ce1d1579fd0af0a77845428cae1b/pyodide-build/pyodide_build/pypabuild.py#L42
to 
```py
            if avoid_name in req.name.lower():
```

**Comment by ocefpaf** [comment]
Should I send a PR or do you prefer to send it?

**Comment by hoodmane** [comment]
I'm not completely sure if this is a correct fix, could you try it and see? I will also test it out. I'm actually a bit confused. It would be great if you would make the PR, but only if this is the right fix.

**Comment by ocefpaf** [comment]
Working on it...

**Comment by hoodmane** [comment]
I don't think this fixes it...

**Comment by hoodmane** [comment]
So it does work to remove `cython` from `UNISOLATED_PACKAGES`. I can't remember why I needed to put it in there right now, so we could try removing it. I suspect the build will fail somewhere but maybe we can find a different fix for that. The comments on #2272 may give some clues.

**Comment by hoodmane** [comment]
> Cython I unisolated specifically because numcodecs needs it but it isn't in the numcodecs build dependencies.

Okay so this is a hack to fix numcodecs. In that case we should really patch numcodes.

**Comment by hoodmane** [comment]
I think the problem is that some packages don't declare that they need cython and their plan is not to recythonize if it isn't present. So we want to include some version of it always.

**Comment by ocefpaf** [comment]
I tested and removing cython from `UNISOLATED_PACKAGES` does work. I understand the conundrum here b/c some projects don't even declare cython as a build requirement. However, as a packager, I believe that is bad practice. We should always re-cythonize. The main reason is probably b/c these packages predates PEP 518, which made it easier to declare a build environment.

TL;DR what is your opinion, patch the packages like `numcodecs` or those who use the lower case `cython`? It is a hard choice that I don't want to make :smile:
